gridstatus/ercot.py

```diff
@@ -2031,36 +2026,182 @@ def _handle_sced_timestamp(self, df, verbose=False):
 
         return df
 
-    def _handle_sced_system_lambda(self, docs, verbose):
-        all_dfs = []
-        for doc in tqdm.tqdm(
+    @support_date_range("DAY_START")
+    def get_dam_shadow_prices(self, date, end=None, verbose=False):
+        """Get Day-Ahead Market Shadow Prices
+
+        Arguments:
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: A DataFrame with day-ahead market shadow prices
+        """
+        # Subtract one day since this is the day ahead market
+        date = date if date == "latest" else date - pd.DateOffset(days=1)
+
+        doc = self._get_document(
+            date=date,
+            report_type_id=DAM_SHADOW_PRICES_RTID,
+            verbose=verbose,
+        )
+
+        return self._handle_dam_shadow_prices_file(doc, verbose=verbose)
+
+    def _handle_dam_shadow_prices_file(self, doc, verbose):
+        df = self.read_doc(doc, parse=True, verbose=verbose)
+        df = df.rename(columns=self._shadow_prices_column_name_mapper())
+
+        df["Publish Time"] = pd.to_datetime(doc.publish_date)
+
+        # This seems to be the same as Interval End
+        df["Delivery Time"] = pd.to_datetime(df["Delivery Time"]).dt.tz_localize(
```

Comment:
Looking at the raw data, it seems "DeliveryTime" is what we call "Interval End", in which case, I think we can drop "Delivery Time."

| DeliveryDate | HourEnding | ConstraintID | ConstraintName    | ContingencyName | ConstraintLimit | ConstraintValue | ViolationAmount | ShadowPrice | FromStation | ToStation | FromStationkV | ToStationkV | DeliveryTime         | DSTFlag | 
|--------------|------------|--------------|-------------------|-----------------|-----------------|-----------------|-----------------|-------------|-------------|-----------|---------------|-------------|----------------------|---------| 
| 02/28/2024   | 01:00      | 12           |  HAMILT_MAVERI1_1 |  SBRAHAM8       |  122            |  122            |  0              |  20.481     |  HAMILTON   |  MAVERICK |  138.10001    |  138        |  02/28/2024 01:00:00 | N       | 
| 02/28/2024   | 01:00      | 11           |  BRUNI_69_1       |  XFOW58         |  37             |  37             |  0              |  953.256    |  BRUNI      |  BRUNI    |  138          |  69         |  02/28/2024 01:00:00 | N       | 
| 02/28/2024   | 24:00      | 1028         |  6036__A          |  DMGSBTR5       |  1340           |  1340           |  0              |  57.563     |  TKWSW      |  MGSES    |  345          |  345        |  02/29/2024 00:00:00 | N       | 
| 02/28/2024   | 24:00      | 1029         |  6520__E          |  DWLFMOS5       |  234            |  234            |  0              |  12.607     |  ODEHV      |  YARBR    |  138          |  138        |  02/29/2024 00:00:00 | N       | 
| 02/28/2024   | 24:00      | 1032         |  FREER_LOBO1_1    |  XFOW58         |  39             |  39             |  0              |  319.952    |  LOBO       |  FREER    |  69           |  69         |  02/29/2024 00:00:00 | N       |